nyxx_lavalink/lib/src/Cluster.dart

```diff
@@ -0,0 +1,258 @@
+part of nyxx_lavalink;
+
+class _Cluster {
+  /// A reference to the client
+  final Nyxx _client;
+
+  /// The client id provided to lavalink;
+  final Snowflake _clientId;
+
+  /// All available nodes, ordered by node id
+  final Map<int, Node> _nodes = {};
+
+  /// Nodes that are currently connecting to server, when a node gets connected
+  /// it will be moved to [_nodes], and when reconnecting will be moved here again
+  final Map<int, Node> _connectingNodes = {};
+
+  /// A map to keep the assigned node id for each player
+  final Map<Snowflake, int> _nodeLocations = {};
+
+  int _lastId = 0;
```

Comment:
what do you mean?

Reply:
why do we need to keep track of id in that variable?

Reply:
we keep track of the last id to avoid repeating node ids as those would be overwritten on the `Map`,
also `_nodeLocations` is used to avoid iterating over all nodes every time a user wants to retrieve a node for a specific guild id